src/VisualStudio/RemoteHostClientMock/Remote/RemoteHostClientFactory.cs

```diff
@@ -15,13 +15,20 @@ internal class RemoteHostClientFactory : IRemoteHostClientFactory
     {
         public Task<RemoteHostClient> CreateAsync(Workspace workspace, CancellationToken cancellationToken)
         {
-            // this is the point where we can create different kind of remote host client in future (cloud or etc)
-            if (workspace.Options.GetOption(RemoteHostClientFactoryOptions.RemoteHost_InProc))
+            try
             {
-                return InProcRemoteHostClient.CreateAsync(workspace, cancellationToken);
-            }
+                // this is the point where we can create different kind of remote host client in future (cloud or etc)
+                if (workspace.Options.GetOption(RemoteHostClientFactoryOptions.RemoteHost_InProc))
+                {
+                    return InProcRemoteHostClient.CreateAsync(workspace, cancellationToken);
+                }
 
-            return ServiceHubRemoteHostClient.CreateAsync(workspace, cancellationToken);
+                return ServiceHubRemoteHostClient.CreateAsync(workspace, cancellationToken);
+            }
+            catch
+            {
+                return Task.FromResult<RemoteHostClient>(null);
```

Comment:
specialized tasks is not available in this proj. and I dont want to take more dependency to this csproj than needed due to all kinds of project.json version conflicts caused by .Next project